android/uhabits-core/src/main/java/org/isoron/uhabits/core/models/Habit.java

```diff
@@ -148,9 +148,9 @@ public synchronized String getDescription()
         return data.description;
     }
 
-    public synchronized void setDescription(@NonNull String description)
+    public synchronized void setDescription(@Nullable String description)
     {
-        data.description = description;
+        data.description = description == null ? "" : description;
```

Comment:
For consistency, I think it would be better to keep this method as `setDescription(@NonNull string description)`, and move the null check to whoever is calling this method. In general, I would like to avoid `nulls` being passed around.